routers/user/home.go

```diff
@@ -216,22 +222,12 @@ func Issues(ctx *context.Context) {
 		if isPullList {
 			unitType = models.UnitTypePullRequests
 		}
-		userRepoIDs, err = ctxUser.GetAccessRepoIDs(unitType)
-		if err != nil {
-			ctx.ServerError("ctxUser.GetAccessRepoIDs", err)
-			return
-		}
+		opts.RepoSubQuery = ctxUser.UnitRepositoriesSubQuery(unitType)
```

Comment:
Later on `opts.RepoIDs` is overwritten in several occasions. `RepoSubQuery` needs to be cleared in some of those cases. For example in:
```
239:		// force an empty result
240:		opts.RepoIDs = []int64{-1}
```

Reply:
I fixed this one with a bigger change.